### PR TITLE
New AVX512F implementation

### PR DESCRIPTION
**Pull request by Ka-zam**
I think this is a better implementation of the reciprocal kernel as it uses the new _mm512_rcp14_ps intrinsic that handles exceptions correctly. It's accurate to tol < 6.2e-5. On a 7950X3D there is a 30% speedup.

```
magnus@r7950x3d:~/src/kazam/volk/build$ volk_profile -R reci
RUN_VOLK_TESTS: volk_32f_reciprocal_32f(131071,1987)
generic completed in 20.7839 ms
a_sse completed in 41.2548 ms
a_avx completed in 20.6385 ms
a_avx512 completed in 16.861 ms
u_sse completed in 41.301 ms
u_avx completed in 20.7819 ms
u_avx512 completed in 15.9916 ms
Best aligned arch: u_avx512
Best unaligned arch: u_avx512
Writing /home/magnus/.volk/volk_config...
```

### CODE REVIEW

File: kernels/volk/volk_32f_reciprocal_32f.h [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+#include <xmmintrin.h>
+static inline void
+volk_32f_reciprocal_32f_a_sse(float* out, const float* in, unsigned int num_points)
+{
+    const __m128 ONE = _mm_set_ps1(1.f);
+    const unsigned int quarter_points = num_points / 4;
+    for (unsigned int number = 0; number < quarter_points; number++) {
+        __m128 x = _mm_load_ps(in);
+        in += 4;
+        __m128 r = _mm_div_ps(ONE, x);
```
Comment (jdemel) [inline kernels/volk/volk_32f_reciprocal_32f.h:77]: does it have the same exception behavior that `1.f / var` has?

File: kernels/volk/volk_32f_reciprocal_32f.h [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+#include <immintrin.h>
+static inline void
+volk_32f_reciprocal_32f_a_avx(float* out, const float* in, unsigned int num_points)
+{
+    const __m256 ONE = _mm256_set1_ps(1.f);
+    const unsigned int eighth_points = num_points / 8;
+    for (unsigned int number = 0; number < eighth_points; number++) {
+        __m256 x = _mm256_load_ps(in);
+        in += 8;
+        __m256 r = _mm256_div_ps(ONE, x);
```
Comment (jdemel) [inline kernels/volk/volk_32f_reciprocal_32f.h:99]: does it have the same exception behavior that 1.f / var has?

**Review by jdemel — COMMENTED** [review]
Thanks for your contribution. It looks really good. We just need to figure out and document if the division handling is the same across all implementations. This seems to be a very nasty to debug bug, if it would occur.

**Comment by Ka-zam** [comment]
I ran all kernels for special values:

```
magnus@r7950x3d:~/src/kazam/scratch$ ./a.out 
x:
 -0.0000e+00   0.0000e+00          inf         -inf          nan         -nan   1.0000e-30   1.0000e+30 
generic:
        -inf          inf   0.0000e+00  -0.0000e+00          nan         -nan   1.0000e+30   1.0000e-30 
a_sse:
        -inf          inf   0.0000e+00  -0.0000e+00          nan         -nan   1.0000e+30   1.0000e-30 
a_avx:
        -inf          inf   0.0000e+00  -0.0000e+00          nan         -nan   1.0000e+30   1.0000e-30 
a_avx512:
        -inf          inf   0.0000e+00  -0.0000e+00          nan         -nan   9.9999e+29   9.9999e-31 
```

NaN and inf with sign are properly handled for all kernels.

**Review by jdemel — APPROVED** [review]
LGTM

**Comment by jdemel** [comment]
No objections. The broken build should be fixed now with #761 . Merging...

**Pull request merged**